rediscluster/pipeline.py

```diff
@@ -1,40 +1,45 @@
 # -*- coding: utf-8 -*-
 
 # python std lib
-import random
 import sys
-import time
 
 # rediscluster imports
 from .client import StrictRedisCluster
 from .exceptions import (
-    RedisClusterException, AskError, MovedError, ClusterDownError, ClusterError
+    RedisClusterException, AskError, MovedError, TryAgainError,
 )
-from .utils import clusterdown_wrapper
+from .utils import clusterdown_wrapper, dict_merge
 
 # 3rd party imports
 from redis import StrictRedis
-from redis.exceptions import ResponseError, ConnectionError, RedisError, TimeoutError
+from redis.exceptions import ConnectionError, RedisError, TimeoutError
 from redis._compat import imap, unicode
 
 
+ERRORS_ALLOW_RETRY = (ConnectionError, TimeoutError, MovedError, AskError, TryAgainError)
+
+
 class StrictClusterPipeline(StrictRedisCluster):
     """
     """
 
-    def __init__(self, connection_pool, result_callbacks=None, reinitialize_steps=None,
-                 response_callbacks=None, startup_nodes=None, refresh_table_asap=False):
+    def __init__(self, connection_pool, result_callbacks=None,
+                 response_callbacks=None, startup_nodes=None):
         """
         """
         self.command_stack = []
         self.connection_pool = connection_pool
-        self.refresh_table_asap = refresh_table_asap
-        self.reinitialize_counter = 0
-        self.reinitialize_steps = reinitialize_steps or 25
         self.response_callbacks = response_callbacks
         self.result_callbacks = result_callbacks
         self.startup_nodes = startup_nodes if startup_nodes else []
 
+        self.refresh_table_asap = False
+        self.nodes_flags = self.__class__.NODES_FLAGS.copy()
+        self.result_callbacks = self.__class__.RESULT_CALLBACKS.copy()
```

Comment:
If we can send in result_callbacks and response_callbacks as vairables and then we just override it anyway with this line, would it not make sense to remove those 2 variables from `__init__` and just set them with this copy command?